config/externalname.go

```diff
@@ -1546,6 +1546,9 @@ var ExternalNameConfigs = map[string]config.ExternalName{
 	//
 	// Batch Scheduling Policy can be imported using the arn: arn:aws:batch:us-east-1:123456789012:scheduling-policy/sample
 	"aws_batch_scheduling_policy": config.TemplatedStringAsIdentifier("name", "arn:aws:batch:{{ .setup.configuration.region }}:{{ .setup.client_metadata.account_id }}:scheduling-policy/{{ .external_name }}"),
+	// Batch Job Definition can be imported using ARN that has a random substring, revision at the end:
+	// arn:aws:batch:us-east-1:123456789012:job-definition/sample:1
+	"aws_batch_job_definition": config.IdentifierFromProvider,
```

Comment:
```suggestion
	"aws_batch_job_definition": config.TemplatedStringAsIdentifier("name", "arn:aws:batch:{{ .setup.configuration.region }}:{{ .setup.client_metadata.account_id }}:job-definition/{{ .external_name }}"),
```
If you haven't tried it, can you please try the configuration above?

Reply:
Thank you for the suggestion @turkenf. That was the first configuration I attempted, which ended up reproducing the behavior described in #162.

In summary, the configuration you suggested would successfully create the external resource but then the controller could not observe it afterwards due to:
> Message: observe failed: cannot run refresh: refresh failed: "name" ("test:1") must be up to 128 letters (uppercase and lowercase), numbers, underscores and dashes, and must start with an alphanumeric.:  

In other words, the external name part of the ARN for a job-definition has a name (defined in config, i.e. `test`) and a version (provided by AWS, i.e. `:1`).

If I understood the [Contributing docs](https://github.com/upbound/upjet/blob/main/docs/add-new-resource-short.md) correctly, the `version` suffix provided by AWS breaks the pattern for [Case 5: Non-random Substrings as Identifier](https://github.com/upbound/upjet/blob/main/docs/add-new-resource-short.md#case-5-non-random-substrings-as-identifier) used for most ARNs and instead falls into the random substring case:
> However, there are cases where the ARN includes random substring and that would fall under [Case 4](https://github.com/upbound/upjet/blob/main/docs/add-new-resource-short.md#case-4-random-identifier-substring-from-provider).

I ended up choosing `config.IdentifierFromProvider` based on other resources with ARNs following the `name:version` pattern, such as an [`aws_ecs_task_definition` external name](https://github.com/upbound/provider-aws/blob/main/config/externalname.go#L379-L381). This worked locally, as I could create and observe the external resource from a local Crossplane cluster (created by `make local-deploy`) against my own AWS account.

I've never worked with golang before and mostly relied on the docs and Makefile to create this PR, so I'm certainly open for guidance on the best way to configure this. I am not sure how to troubleshoot the automated test logs or the error (`uptest-v0.5.0: error: cannot run e2e tests successfully: cannot execute tests: kuttl failed: exit status 1`) and can't seem to get `make e2e` run on my machine. Any help is much appreciated!